### DISCUSSION TITLE
Autolock does not activate on sleep

### DISCUSSION THREAD

**Discussion by sethmad** [standardnotes/forum#2564] [Suggestions]
In the desktop app, autolock doesn't activate when the machine goes to sleep. If you wake the machine before the autolock time has expired, the app loads unlocked. I would expect the app to lock on sleep.

**To Reproduce**
1. Open the Mac desktop app with passcode lock enabled and autolock set to a duration other than "immediately."
2. Put the computer to sleep using the File menu
3. Wake up the machine, log back in before the autolock time limit
4. App is unlocked

 - OS: OS X 11.1 (Intel machine)
 - Standard Notes desktop 3.6.8

**Comment by JaspalSuri** [comment]
Hi @sethmad, thank you for bringing this issue to our attention! I'll look into this and forward it over to the team for further review.

**Comment by JaspalSuri** [comment]
Hello again @sethmad, I'm afraid that the app currently cannot detect when your system goes to sleep, as it might be frozen before it can respond, so it wouldn't be able to autolock prematurely. If this becomes possible in the future, we'll consider implementing this enhancement. 🙂
